### PR TITLE
Updated to use 4.0.0 version of BSON.

### PR DESCRIPTION
**Pull request by mpokress**
Still waiting on @alethea's fix to make it into the released BSON, but this version already does a better job avoiding collisions.

https://pillpack.atlassian.net/browse/PD-3810

### CODE REVIEW

File: object_id_primary_key.gemspec [hunk 1]
```diff
@@ -17,8 +17,9 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^bin/}) { |f| File.basename(f) }
   spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
   spec.require_paths = ["lib"]
+  spec.required_ruby_version = '~> 2'
 
-  spec.add_dependency 'bson', '~> 2.3.0'
+  spec.add_dependency 'bson', '~> 4.0.0'
```
Comment (alethea) [inline object_id_primary_key.gemspec]: We should change this to `'~> 4.0'`, BSON conforms to semantic versioning. Don't want to get us locked down again.

Comment (mpokress) [inline object_id_primary_key.gemspec]: Yeah, dumb. I meant to do 4.0 so we get all that 4.0.1 goodness you requested from them.

Comment (mpokress) [inline object_id_primary_key.gemspec]: Wait, why does that matter? This is gte 4.0.0. Won't that get us any 4.x.y?

When I ran bundler locally this morning it was pulling down rails 4.2.1 stuff. Admittedly, I thought that was weird that it wasn't getting 4.2.5. Didn't pursue it b/c it wasn't relevant to the tests.

Comment (alethea) [inline object_id_primary_key.gemspec]: No, `'~> 4.0.0'` will get us any 4.0.x, `'~> 4.0`' will get us any 4.x.y

Comment (mpokress) [inline object_id_primary_key.gemspec]: :skull: :boom: 

(head explodes from confusion)

**Comment by alethea** [comment]
Looks good to me.

**Pull request merged**